bindsnet/network/topology.py

```diff
@@ -148,12 +148,12 @@ def __init__(self, source: Nodes, target: Nodes, nu: Optional[Union[float, Seque
 
         self.w = kwargs.get('w', None)
         if self.w is None:
-            if self.wmin is None or self.wmax is None:
+            if self.wmin == -np.inf or self.wmax == np.inf:
```

Comment:
What is one is finite and the other isn't? E.g., if `self.wmax` is 0.5, `torch.rand()` may give values larger than it.